### PR TITLE
Add STS dataset SemRel2024

### PR DESCRIPTION
**Pull request by akshita-sukhlecha**
### Adding datasets checklist

- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 

Also tested model `sentence-transformers/LaBSE` and validated that the results matched with the numbers given in [dataset paper Table 12](https://arxiv.org/pdf/2402.08638)

### CODE REVIEW

File: mteb/tasks/STS/multilingual/SemRel24STS.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    "eng": ["eng-Latn"],
+    "hau": ["hau-Latn"],
+    "hin": ["hin-Deva"],
+    "ind": ["ind-Latn"],
+    "kin": ["kin-Latn"],
+    "mar": ["mar-Deva"],
+    "tel": ["tel-Telu"],
+}
+
+_SPLITS = ["dev", "test"]
```
Comment (KennethEnevoldsen) [inline mteb/tasks/STS/multilingual/SemRel24STS.py:20]:
```suggestion
_SPLITS = ["test"]
```
test seems big enough (this makes the benchmark a bit faster to run.

File: docs/mmteb/points/917.jsonl
Comment (KennethEnevoldsen) [inline docs/mmteb/points/917.jsonl:1]: We have actually stopped adding points for additional datasets (with a few exceptions). I would be ok with adding an exception here as it adds quite a few languages to STS (@imenelydiaker just checking with you as well)

Will you add points for review as well?

**Review by KennethEnevoldsen — APPROVED** [review]

File: docs/mmteb/points/917.jsonl
Comment (akshita-sukhlecha) [reply docs/mmteb/points/917.jsonl:1]: @KennethEnevoldsen @imenelydiaker Oh okay. I got to know of this MMTEB effort at the beginning of this month itself and had been doing similar stuff for Indic languages for a project. 

Let me know if its okay to keep the points. Accordingly update and merge the PR.

Comment (KennethEnevoldsen) [reply docs/mmteb/points/917.jsonl:1]: This is reasonable as it stands. Will merge it (if @imenelydiaker disagrees, we might change it)

**Pull request merged**